### PR TITLE
Build a Gatsby Blog using the Cosmic JS source plugin

### PR DESCRIPTION
**Pull request by tonyspiro**
Gatsby 🔥 + Cosmic 🛠 🚀

### CODE REVIEW

**Comment by pieh** [comment]
Preview of blog post: https://deploy-preview-5783--gatsbyjs.netlify.com/blog/2018-06-07-build-a-gatsby-blog-using-the-cosmic-js-source-plugin/

**Comment by jlengstorf** [comment]
@tonyspiro Thanks so much for sending this over! I just went through and made some edits to the post (see my commit message above for details), but overall this looks like a great intro to CosmicJS!

A quick clarification: in the tutorial, the source plugin config has a field for an API key. Is there a link or instructions for getting one of those keys? Do I need one to get data loaded from CosmicJS? Will my local blog have data if I don't create a CosmicJS account and add some blogs beforehand?

If this is a special demo so people don't need to create an account, it may be worth adding some clarification on that, because I found myself wondering if this tutorial would work for me without first creating a bunch of content to work with.

**Comment by tonyspiro** [comment]
Thank you @jlengstorf!  Your edits look good to me.

1. For the API key, this is optional and can be set in the Cosmic JS Bucket dashboard.  I can add a line to indicate where to do this (Your Cosmic JS Bucket > Settings > Basic Settings).

2. As for the blog content, if you download the full GitHub repo, it defaults a connection to a demo Bucket with all of the demo data and it will work right out-of-the-box.  To be able to edit content, you would need to install the app (with demo content) and config in the code to connect to this Bucket.  I can add a line to explain this as well.

Let me know if you want me to add these.  Thanks!

**Comment by jlengstorf** [comment]
Hey, @tonyspiro — yes, please do! I think it will help keep things clear for readers. Thanks!

**Comment by tonyspiro** [comment]
Hey @jlengstorf, changes pushed. Let me know if you need anything else, thanks!

**Comment by jlengstorf** [comment]
@tonyspiro This looks great!

@shannonbux has been managing the blog publishing schedule, so I'll tag her in here for any remaining edits and/or pushing the button to take it live. 😄

**Comment by tonyspiro** [comment]
Thank you @jlengstorf! And thank you @shannonbux, let me know if you need anything.

File: docs/blog/2018-06-07-build-a-gatsby-blog-using-the-cosmic-js-source-plugin/index.md [hunk 1]
```diff
@@ -0,0 +1,424 @@
+---
+title: Build a Gatsby Blog using the Cosmic JS source plugin
+date: "2018-06-07"
+author: "Tony Spiro"
```
Comment (KyleAMathews) [inline docs/blog/2018-06-07-build-a-gatsby-blog-using-the-cosmic-js-source-plugin/index.md]: Can you add `canonicalLink` here to your original post so you get to keep the google juice ;-) https://github.com/gatsbyjs/gatsby/blob/82d9188aba871bb0bb94e32c3e590cb13ae3cd12/docs/blog/2018-04-10-how-to-handle-comments-in-gatsby-blogs/index.md#L5

Comment (tonyspiro) [reply docs/blog/2018-06-07-build-a-gatsby-blog-using-the-cosmic-js-source-plugin/index.md]: Thank you @KyleAMathews! This is done.

**Comment by jlengstorf** [comment]
Hiya @tonyspiro! 👋

This is definitely late, but on behalf of the entire Gatsby community, I wanted to say thank you for being here.

Gatsby is built by awesome people like you. Let us say “thanks” in two ways:

1.  **We’d like to send you some Gatsby swag.** As a token of our appreciation, you can go to the [Gatsby Swag Store][store] and log in with your GitHub account to get a coupon code good for one free piece of swag. (We’ve got t-shirts and hats, plus some socks that are really razzing our berries right now.)
2.  **If you’re not already part of it, we just invited you to join the Gatsby organization on GitHub.** This will add you to our team of maintainers. You’ll receive an email shortly asking you to confirm. By joining the team, you’ll be able to label issues, review pull requests, and merge approved pull requests.

If you have questions, please don’t hesitate to reach out to us: tweet at [@gatsbyjs][twitter] and we’ll come a-runnin’.

Thanks again! 💪💜

[store]: https://store.gatsbyjs.org
[twitter]: https://twitter.com/gatsbyjs

**Pull request merged**